File: dailyge-api/src/main/java/project/dailyge/app/common/configuration/QueryDslConfig.java
package project.dailyge.app.common.configuration;

import com.querydsl.jpa.impl.*;
import jakarta.persistence.*;
import org.springframework.context.annotation.*;

@Configuration
public class QueryDslConfig {

    @PersistenceContext
    private EntityManager entityManager;

    @Bean
    public JPAQueryFactory jpaQueryFactory() {
        return new JPAQueryFactory(entityManager);
    }
}

File: storage/rdb/src/main/kotlin/project/dailyge/configuration/QueryDslConfig.kt
package project.dailyge.configuration

import com.querydsl.jpa.impl.JPAQueryFactory
import jakarta.persistence.EntityManager
import jakarta.persistence.PersistenceContext
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class QueryDslConfig {

    @PersistenceContext
    private lateinit var entityManager: EntityManager

    @Bean
    fun jpaQueryFactory(): JPAQueryFactory = JPAQueryFactory(entityManager)
}
